### DISCUSSION TITLE
Does ComfyUI have an accessible module to run diffusion forward? (i.e. noising rather than de-noising)

### DISCUSSION THREAD

**Discussion by ttulttul** [Comfy-Org/ComfyUI#2206] [Q&A]
I'd like to take a crack at implementing a custom node for [DemoFusion](https://arxiv.org/pdf/2311.16973.pdf) (new upscaling technique). To do so, I need to run checkpoint models in the forward (noising) direction. I can see Comfy has its own implementation of models and that they have a forward() call, but I'm not certain how you would access this appropriately from a node. Has anyone done this before?

**Comment by yorkb3d** [comment]
https://youtu.be/Ev44xkbnbeQ?si=dJ3jiV8S5DbpBQow
The unsampler node at 9:30 in this video. Is this what you were after?

**Reply by ttulttul** [reply]
Thanks. I was able to use the Unsampler node to try out some concepts from the paper; however, given the number of steps of noising and denoising required for the algorithm described in the paper, I think this has to be implemented using a custom node that can access Comfy's sampler routines directly. I've read that code and ... well, if I have time I'll give it a shot this week.

**Reply by yorkb** [reply]
Good luck with it. It would be great to see this implemented in Comfy.

**Comment by ltdrdata** [comment]
This seems like a current PoC level of implementation, but it looks like a good starting point from here.

https://github.com/deroberon/demofusion-comfyui

**Reply by ttulttul** [reply]
Fantastic. It seems he pulled in the official library and just hacked it into a node. A great starting point.
